app/controllers/users_controller.rb

```diff
@@ -0,0 +1,101 @@
+class UsersController < ApplicationController
+  skip_before_action :require_login, only: [:upvote] #g
```

Comment:
I'm pretty sure you meant for this to be a `before_action` and not a `skip_before_action`.

```suggestion
  before_action :require_login, only: [:upvote] #g
```